modules/packages/ArgumentParser.chpl

```diff
@@ -425,19 +507,69 @@ module ArgumentParser {
     }
  }
 
+  /*
+  A parser that performs the following functions:
+
+    1. Stores definitions of valid arguments/flags/options
+    2. Parses an array of arguments as passed from the command line
+       (where args[0] is typically the name of the executable)
+    3. Throws an ArgumentError if invalid or unrecognized arguments are encountered
+
+  The parser can process command line arguments that include or omit an equals
+  sign ``=``, with short ``-`` or long ``--`` indicators.
+
+  We can declare a new argumentParser like so:
+
+  .. code-block:: chapel
+
+    var parser = new argumentParser();
+
+  **Limitations**
+
+   * Cannot place short flags together. That is, if a parser has defined short
+     flags ``-c``, ``-a`` and ``-t``, we cannot handle ``-cat``.
+
+   * Short options must be separated from their values with an equals sign or
+     spaces. That is, ``-cValue`` is not supported and should be entered as either
+     ``-c Value`` or ``-c=Value``.
+
+   * Numeric types are not supported and will be passed as strings.
+
+   * Negative numbers on the command line may cause issues.
+
+   * Shadowing config vars or Chapel arguments can be accomplished by placing
+     those values after a ``--`` on the command line. If a passthrough is also
+     specified as ``--``, there will be ambiguity and your shadowed arguments
+     will be captured by the passthrough. As a workaround, specify a custom
+     delimiter for passthrough.
+
+   * Help and usage are not automatically generated or handled internally. This
+     is planned for future development.
+
+   * Counting the number of times a flag is found on the command line is not
+     supported.
+
+   * Mutual exclusion or conditionally required arguments are not supported.
```

Comment:
Maybe say it this way:

"The argument parser doesn't support checking for mutually exclusive arguments or arguments that are conditionally required. Instead, you can check after parsing the arguments that the provided arguments are acceptable."